oauth/resource_service_principal_federation_policy.go

```diff
@@ -0,0 +1,147 @@
+package oauth
+
+import (
+	"context"
+	"github.com/databricks/databricks-sdk-go/service/oauth2"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"log"
+	"strings"
+)
+
+func ResourceServicePrincipalFederationPolicy() common.Resource {
+	s := map[string]*schema.Schema{
```

Comment:
As I understand it, the contribution guide asks that the schema is created using `common.StructToSchema`.

https://github.com/databricks/terraform-provider-databricks/blob/main/CONTRIBUTING.md#adding-a-new-resource

For example:

https://github.com/databricks/terraform-provider-databricks/blob/c6f6f77935fe4553c491c53316b59187764fdf7d/tokens/resource_service_principal_secret.go#L27

This looks like the correct struct to use:

https://github.com/databricks/terraform-provider-databricks/blob/c6f6f77935fe4553c491c53316b59187764fdf7d/internal/service/oauth2_tf/model.go#L994